### PR TITLE
[core] Clean error messages prefix

### LINKED ISSUE #2631: Material-UI -> MUI renaming in the codebase
There are multiple instances where we can rename the new project's name

https://github.com/mui-org/material-ui-x/blob/e9a9ecc3bf932d38c8a1cf5edbd1abc8eb3de11d/packages/grid/data-grid/src/DataGrid.tsx#L88

https://github.com/mui-org/material-ui-x/blob/e9a9ecc3bf932d38c8a1cf5edbd1abc8eb3de11d/packages/grid/data-grid/package.json#L13

### PR DESCRIPTION
**Pull request by flaviendelangle**
Closes #2631

### CODE REVIEW

**Comment by m4theushw** [comment]
Do you plan to also update the documentation?

**Comment by oliviertassinari** [comment]
In the issue description, I didn't try to provide a comprehensive list of the type of references that needs to be updated. I would encourage to do a global search for `Material-UI` and material-ui.com.

**Comment by flaviendelangle** [comment]
I can update this pr with the other cases yes

File: docs/pages/api-docs/data-grid/data-grid-pro.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ The name <code>MuiDataGrid</code> can be used when providing [default props](/cu
 | <span class="prop-name">isCellEditable</span> | <span class="prop-type">(params: GridCellParams) => boolean</span> |   | Callback fired when a cell is rendered, returns `true` if the cell is editable. |
 | <span class="prop-name">isRowSelectable</span> | <span class="prop-type">(params: GridRowParams) => boolean</span> |   | Determines if a row can be selected. |
 | <span class="prop-name">loading</span> | <span class="prop-type">boolean</span> |  false | If `true`, a  loading overlay is displayed.. |
-| <span class="prop-name">localeText</span> | <span class="prop-type">GridLocaleText</span> |   | Set of text labels used in the grid. You can find all the translation keys supported in [the source](https://github.com/mui-org/material-ui-x/blob/HEAD/packages/grid/_modules_/grid/constants/localeTextConstants.ts) in the GitHub repository. |
+| <span class="prop-name">localeText</span> | <span class="prop-type">GridLocaleText</span> |   | Set of text labels used in the grid. You can find all the translation keys supported in [the source](https://github.com/mui-org/MUI-x/blob/HEAD/packages/grid/_modules_/grid/constants/localeTextConstants.ts) in the GitHub repository. |
```
Comment (oliviertassinari) [inline docs/pages/api-docs/data-grid/data-grid-pro.md:65]:
```suggestion
| <span class="prop-name">localeText</span> | <span class="prop-type">GridLocaleText</span> |   | Set of text labels used in the grid. You can find all the translation keys supported in [the source](https://github.com/mui-org/material-ui-x/blob/HEAD/packages/grid/_modules_/grid/constants/localeTextConstants.ts) in the GitHub repository. |
```

File: docs/pages/api-docs/data-grid/data-grid-pro.md [hunk 2]
```diff
@@ -132,7 +132,7 @@ You can use the [slots API](/components/data-grid/components/#overriding-compone
 
 | Name | Type | Default | Description |
 |:-----|:-----|:--------|:------------|
-| <span class="prop-name">Checkbox</span> | <span class="prop-type">elementType</span> | <span class="prop-type">Checkbox</span> | Checkbox component used in the grid for both header and cells. By default, it uses the Material-UI core Checkbox component. |
+| <span class="prop-name">Checkbox</span> | <span class="prop-type">elementType</span> | <span class="prop-type">Checkbox</span> | Checkbox component used in the grid for both header and cells. By default, it uses the MUI core Checkbox component. |
```
Comment (oliviertassinari) [inline docs/pages/api-docs/data-grid/data-grid-pro.md:135]:
```suggestion
| <span class="prop-name">Checkbox</span> | <span class="prop-type">elementType</span> | <span class="prop-type">Checkbox</span> | Checkbox component used in the grid for both header and cells. By default, it uses the MUI core Checkbox component. |
```

File: README.md [hunk 3]
```diff
@@ -26,26 +26,26 @@ We provide three options:
...
-- [DataGridPro](https://material-ui.com/components/data-grid/#commercial-version), published under `@mui/x-data-grid-pro`, commercially licensed.
+- [DataGrid](https://mui.com/components/data-grid/), published under `@mui/x-data-grid`, MIT licensed.
+- [DataGridPro](https://mui.com/components/data-grid/#commercial-version), published under `@mui/x-data-grid-pro`, commercially licensed.
 
 ## Questions
 
 For _how-to_ questions and other non-issues,
-please use [StackOverflow](https://stackoverflow.com/questions/tagged/material-ui) instead of GitHub issues.
-There is a StackOverflow tag called "material-ui" that you can use to tag your questions.
+please use [StackOverflow](https://stackoverflow.com/questions/tagged/mui) instead of GitHub issues.
```
[the remark this answers is not in the document]
Comment (flaviendelangle) [reply README.md:35]: We have a redirection from https://stackoverflow.com/questions/tagged/mui to https://stackoverflow.com/questions/tagged/material-ui no ?
I tried it

File: docs/pages/api-docs/data-grid/data-grid-pro.md (hunk 2 above)
Comment (flaviendelangle) [reply docs/pages/api-docs/data-grid/data-grid-pro.md:135]: What is the difference between the two lines here ?

Comment (oliviertassinari) [reply docs/pages/api-docs/data-grid/data-grid-pro.md:135]: It was the same oops. I wondered about the capitalization of the MUI Core.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryancogswell) [reply README.md:35]: I looked into this some right after the 5.0 release. I think it should be fine to suggest that people use the `mui` tag. The tags will be automatically renamed to `material-ui`, but that rename is tracked:
image

Once the number of renames becomes more significant, we can submit a request to StackOverflow to invert the relationship to make `mui` the master and `material-ui` the synonym. I know how to make that request, but I don't know how long the process takes. I'm pretty sure they are unlikely to invert an alias relationship prior to a significant change in the trend of which tag people try to use when creating questions.

[the remark this answers is not in the document]
Comment (oliviertassinari) [reply README.md:35]: Awesome, ok so we can actively encourage the use of `mui`. cc @mnajdova has it's also impacting the main repository.

File: docs/pages/api-docs/data-grid/data-grid-pro.md (hunk 2 above)
Comment (oliviertassinari) [reply docs/pages/api-docs/data-grid/data-grid-pro.md]: If we approved this one, could we update the whole codebase? I assume that the source of truth for this one is in the TypeScript' descriptions.

Comment (m4theushw) [reply docs/pages/api-docs/data-grid/data-grid-pro.md]: > If we approved this one, could we update the whole codebase? I assume that the source of truth for this one is in the TypeScript' descriptions.

I think it's not worth it to waste time updating the manually authored API pages now. The documentation for this component can be automatically generated in #2465. BTW, the TypeScript definition doesn't have the "By default, it uses the MUI..." part.

Comment (oliviertassinari) [reply docs/pages/api-docs/data-grid/data-grid-pro.md]: > BTW, the TypeScript definition doesn't have the "By default, it uses the MUI..." part.

Oh nice, +1 for ignoring the api .md files and focusing on the TypeScript definition. It started to diverge on this instance with #1528.

File: packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts [hunk 4]
```diff
@@ -135,7 +135,7 @@ export const useGridKeyboardNavigation = (
             : -1 * containerSizes!.viewportPageSize);
         nextCellIndexes = { colIndex, rowIndex: nextRowIndex };
       } else {
-        throw new Error('Material-UI. Key not mapped to navigation behavior.');
+        throw new Error('MUI. Key not mapped to navigation behavior.');
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts:138]:
```suggestion
        throw new Error('MUI: Key not mapped to navigation behavior.');
```

File: packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts [hunk 5]
```diff
@@ -197,7 +197,7 @@ export const useGridKeyboardNavigation = (
         }
         return;
       } else {
-        throw new Error('Material-UI. Key not mapped to navigation behavior.');
+        throw new Error('MUI. Key not mapped to navigation behavior.');
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/hooks/features/keyboard/useGridKeyboardNavigation.ts:200]:
```suggestion
        throw new Error('MUI: Key not mapped to navigation behavior.');
```

File: packages/grid/_modules_/grid/hooks/root/useGridApiContext.ts [hunk 6]
```diff
@@ -7,7 +7,7 @@ export function useGridApiContext() {
   if (apiRef === undefined) {
     throw new Error(
       [
-        'Material-UI X: Could not find the data grid context.',
+        'MUI X: Could not find the data grid context.',
```
Comment (m4theushw) [inline packages/grid/_modules_/grid/hooks/root/useGridApiContext.ts:10]:
```suggestion
        'MUI: Could not find the data grid context.',
```

File: packages/grid/data-grid/LICENSE [hunk 7]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 Material-UI SAS
+Copyright (c) 2020 MUI SAS
```
Comment (m4theushw) [inline packages/grid/data-grid/LICENSE:3]: The legal entity didn't change.

```suggestion
Copyright (c) 2021 Material-UI SAS
```

x other copyright warnings.

File: packages/grid/data-grid/src/tests/components.DataGrid.test.tsx [hunk 8]
```diff
@@ -65,7 +65,7 @@ describe('<DataGrid /> - Components', () => {
       );
       // @ts-expect-error need to migrate helpers to TypeScript
     }).toErrorDev([
-      'Material-UI X: Could not find the data grid context.',
+      'MUI X: Could not find the data grid context.',
```
Comment (m4theushw) [inline packages/grid/data-grid/src/tests/components.DataGrid.test.tsx:68]:
```suggestion
      'MUI: Could not find the data grid context.',
```

File: packages/x-license/src/licenseErrorMessageUtils.ts [hunk 9]
```diff
@@ -14,31 +14,31 @@ function showError(message: string[]) {
 
 export function showInvalidLicenseError() {
   showError([
-    'Material-UI X: Invalid license.',
+    'MUI X: Invalid license.',
```
Comment (m4theushw) [inline packages/x-license/src/licenseErrorMessageUtils.ts:17]: @oliviertassinari Should the "MUI" prefix be used for license errors?

```suggestion
    'MUI: Invalid license.',
```

x other cases in this file.

File: docs/pages/api-docs/data-grid/data-grid-pro.md [hunk 10]
```diff
@@ -228,7 +228,7 @@ You can override the style of the component thanks to one of these customization
 - With a [global class name](/customization/components/#overriding-styles-with-global-class-names).
 - With a theme and an [`overrides` property](/customization/globals/#css).
 
-If that's not sufficient, you can check the [implementation of the component style](https://github.com/mui-org/material-ui-x/blob/master/packages/grid/_modules_/grid/components/containers/GridRootStyles.ts) for more detail.
+If that's not sufficient, you can check the [implementation of the component style](https://github.com/mui-org/MUI-x/blob/master/packages/grid/_modules_/grid/components/containers/GridRootStyles.ts) for more detail.
```
Comment (m4theushw) [inline docs/pages/api-docs/data-grid/data-grid-pro.md:231]: 404

```suggestion
If that's not sufficient, you can check the [implementation of the component style](https://github.com/mui-org/material-ui-x/blob/next/packages/grid/_modules_/grid/components/containers/GridRootStyles.ts) for more detail.
```

Could you also point to next below?

https://github.com/mui-org/material-ui-x/blob/71cc0d3e6471cd6068a8fba555620953c553a377/docs/pages/api-docs/data-grid/data-grid.md#L219

**Review by m4theushw — APPROVED** [review]

File: packages/grid/x-grid-data-generator/LICENSE [hunk 11]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 Material-UI SAS
+Copyright (c) 2021 Material-UI SAS
```
Comment (oliviertassinari) [inline packages/grid/x-grid-data-generator/LICENSE:3]: AFAIK, this change is not correct, see https://opensource.stackexchange.com/questions/6389/how-do-the-years-specified-in-a-copyright-statement-work for why. It's meant to signal the first year when the content was created.

@mbrookes A preference on what we should do? Like React or

```suggestion
Copyright (c) Material-UI SAS
```

or as before, the first year

```suggestion
Copyright (c) 2020 Material-UI SAS
```

or a range

```suggestion
Copyright (c) 2020-2021 Material-UI SAS
```

?

I would be leaning to revert, match what we do in the core.

File: docs/pages/playground.js [hunk 12]
```diff
@@ -0,0 +1,5 @@
+import * as React from 'react';
+
+export default function Playground() {
+  return <div>This file is listed in `.gitignore`</div>;
+}
```
Comment (oliviertassinari) [inline docs/pages/playground.js:5]: It would have been better as a tsx file.

**Pull request merged**